include/detect_breakends/junction_detection.hpp

```diff
@@ -3,13 +3,9 @@
 #include <seqan3/io/sequence_file/all.hpp>  // FASTA support
 #include <seqan3/io/alignment_file/all.hpp> // SAM/BAM support
 
-#include "detect_breakends/bam_functions.hpp"
-#include "junction.hpp"
-#include "detect_breakends/aligned_segment.hpp"
+#include "detect_breakends/bam_functions.hpp"   // for hasFlag* functions
+#include "junction.hpp"                         // for class junction
+#include "detect_breakends/aligned_segment.hpp" // for struct aligned_segment
 
-/*! \brief Function, detecting junctions in alignment files (sam/bam).
- *  \param alignment_file_path input file - path to the sam/bam file
- *  \param insertion_file_path output file - path for the fasta file
- */
```

Comment:
I wanted to avoid copy & paste comments, so I used `copydoc` in the hpp file, but got warnings while creating the doc.
```bash
.../iGenVar/include/find_deletions/deletion_finding_and_printing.hpp:10: warning: Found recursive @copydetails or @copydoc relation for argument 'find_and_print_deletions()'.

.../iGenVar/include/find_deletions/deletion_finding_and_printing.hpp:16: warning: argument 'junction_file_path' from the argument list of find_and_print_deletions has multiple @param documentation sections
.../iGenVar/include/find_deletions/deletion_finding_and_printing.hpp:10: warning: Found recursive @copydetails or @copydoc relation for argument 'find_and_print_deletions()'.
```

Reply:
@simonsasse That was a really good comment. I checked again about commenting on cpp and or hpp files and have now created a [coding guide](https://github.com/seqan/iGenVar/wiki/page_coding_guide) for this. That is of course welcome to discuss.

Reply:
Gooding guide looks nice. Didn't know that this is really a topic people discuss about ;)